enterprise/server/remote_execution/blockio/blockio.go

```diff
@@ -446,6 +452,14 @@ func (s *COWStore) calculateChunkSize(startOffset int64) int64 {
 }
 
 func (s *COWStore) copyChunk(chunkStartOffset int64) (err error) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
```

Comment:
I think it will hurt performance to have this whole func locked since the goroutines in MergeDiffSnapshot will be blocked waiting for the full copy to complete. We probably want just the map accesses to be locked

Reply:
Done